### PR TITLE
Added TestPlugins, TestRestart and TestRestoreBackup

### PR DESCRIPTION
**Pull request by SylwiaBrant**
Added tests for plugins in Configuration test, tests for restarts and test of backup and restore.

### CODE REVIEW

File: test/e2e/restart_test.go [hunk 1]
```diff
@@ -97,18 +48,19 @@ func checkIfAuthorizationStrategyUnsecuredIsSet(t *testing.T, jenkinsClient jenk
...
+	By("checking that Base Configuration's complete time is not set")
+
 	jenkinsStatus := &v1alpha2.Jenkins{}
 	namespaceName := types.NamespacedName{Namespace: jenkins.Namespace, Name: jenkins.Name}
-	err := framework.Global.Client.Get(context.TODO(), namespaceName, jenkinsStatus)
+	err := k8sClient.Get(context.TODO(), namespaceName, jenkinsStatus)
 	if err != nil {
-		t.Fatal(err)
+		Fail(fmt.Sprintf("Couldn't find jenkinsStatus in namespace '%+v'", namespaceName))
 	}
```
Comment (tomaszsek) [inline test/e2e/restart_test.go:62]: `Expect(err).NotTo(HaveOccurred())`

File: test/e2e/restorebackup_test.go [hunk 2]
```diff
@@ -130,10 +95,48 @@ func createJenkinsWithBackupAndRestoreConfigured(t *testing.T, name, namespace s
...
+									Port:   intstr.FromString("http"),
+									Scheme: corev1.URISchemeHTTP,
+								},
+							},
+							InitialDelaySeconds: int32(80),
+							TimeoutSeconds:      int32(4),
+							FailureThreshold:    int32(10),
+							SuccessThreshold:    int32(1),
+							PeriodSeconds:       int32(1),
+						},
```
Comment (tomaszsek) [inline test/e2e/restorebackup_test.go:139]: please remove, the default provides by the operator should be used

File: test/e2e/seedjobs_test.go [hunk 3]
```diff
@@ -292,3 +293,27 @@ for (BuildStep step : jobRef.getBuildersList()) {
...
+
+	By("running Jenkins job")
+	_, err = job.InvokeSimple(map[string]string{})
+	Expect(err).To(BeNil())
+
+	// FIXME: waitForJobToFinish use
+	By("waiting for the job to finish")
+	time.Sleep(100 * time.Second) // wait for the build to complete
+}
+func verifyJobIsWorkingCorrectly(jenkinsClient jenkinsclient.Jenkins, jobID string) {
```
Comment (tomaszsek) [inline test/e2e/seedjobs_test.go:310]:
```suggestion
func verifyJobIHasBeenRunCorrectly(jenkinsClient jenkinsclient.Jenkins, jobID string) {
```

File: test/e2e/suite_test.go [hunk 4]
```diff
@@ -118,7 +118,7 @@ var _ = BeforeSuite(func(done Done) {
 	k8sClient = k8sManager.GetClient()
 	Expect(k8sClient).NotTo(BeNil())
 	close(done)
-}, 60)
+}, 2000)
```
Comment (tomaszsek) [inline test/e2e/suite_test.go:121]: Why it is so high?

**Review by tomaszsek — CHANGES_REQUESTED** [review]

File: test/e2e/restorebackup_test.go (hunk 2 above)
Comment (SylwiaBrant) [reply test/e2e/restorebackup_test.go:139]: If I do there is a loop of setting new probes until everything crashes. I wonder if it is a problem beyond test?

File: test/e2e/suite_test.go (hunk 4 above)
Comment (SylwiaBrant) [reply test/e2e/suite_test.go:121]: All tests took 1564 s. I can try different values of waiting times in all tests and in probes to make it shorter, but some tests fail, that's why I had to increase times in probes of the main jenkins creating function (not the one from the backup and restore).

File: test/e2e/restorebackup_test.go (hunk 2 above)
Comment (tomaszsek) [reply test/e2e/restorebackup_test.go:139]: Maybe it is a problem with setting defaults. Please look at my commits I've made some changes related to it.

**Pull request merged**